### ISSUE TITLE
strange dots interaction on noteheads

### ISSUE THREAD

**Issue by rettinghaus** [rism-digital/verovio#655]
When dots are added to 'merged' note heads layout gets messed up. In those cases dots should at least merge, too (double-struck). They must never look like double dots! 

![dots](https://user-images.githubusercontent.com/7693447/28065842-c22a46f2-663a-11e7-8283-4edaa4b03d83.png)

see issues #244 #559 #595 on this.

NB: note that double encoded double bar lines are doubled (#364). 

```
<?xml version="1.0" encoding="UTF-8"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
  <meiHead>
    <fileDesc>
      <titleStmt>
        <title />
      </titleStmt>
      <pubStmt />
    </fileDesc>
  </meiHead>
  <music>
    <body>
      <mdiv>
        <score>
          <scoreDef>
            <staffGrp>
              <staffDef clef.shape="G" clef.line="2" n="1" lines="5" />
            </staffGrp>
          </scoreDef>
          <section>
            <measure>
              <staff n="1">
                <layer n="1">
                  <note pname="b" oct="4" dur="2"/>
                </layer>
                <layer n="2">
                  <note pname="b" oct="4" dur="2"/>
                </layer>
              </staff>
            </measure>
            <measure>
              <staff n="1">
                <layer n="1">
                  <note pname="b" oct="4" dur="4"/>
                </layer>
                <layer n="2">
                  <note pname="b" oct="4" dur="4"/>
                </layer>
              </staff>
            </measure>
            <measure right="dbl">
              <staff n="1">
                <layer n="1">
                  <note pname="b" oct="4" dur="8"/>
                </layer>
                <layer n="2">
                  <note pname="b" oct="4" dur="8"/>
                </layer>
              </staff>
            </measure>
            <measure left="dbl">
              <staff n="1">
                <layer n="1">
                  <note pname="b" oct="4" dur="2" dots="1"/>
                </layer>
                <layer n="2">
                  <note pname="b" oct="4" dur="2" dots="1"/>
                </layer>
              </staff>
            </measure>
            <measure>
              <staff n="1">
                <layer n="1">
                  <note pname="b" oct="4" dur="4" dots="1"/>
                </layer>
                <layer n="2">
                  <note pname="b" oct="4" dur="4" dots="1"/>
                </layer>
              </staff>
            </measure>
            <measure>
              <staff n="1">
                <layer n="1">
                  <note pname="b" oct="4" dur="8" dots="1"/>
                </layer>
                <layer n="2">
                  <note pname="b" oct="4" dur="8" dots="1"/>
                </layer>
              </staff>
            </measure>
          </section>
        </score>
      </mdiv>
    </body>
  </music>
</mei>
```

**Comment by lpugin** [comment]
Fixed

**Comment by craigsapp** [comment]
This is what the current `develop` branch version looks like:

<img width="904" alt="screen shot 2018-04-25 at 6 49 30 am" src="https://user-images.githubusercontent.com/3487289/39250081-027604ce-4855-11e8-82b1-d0fd631357e5.png">

There are still two dots instead of one dot on the last notes:

```xml
           <measure>
              <staff n="1">
                <layer n="1">
                  <note pname="b" oct="4" dur="8" dots="1"/>
                </layer>
                <layer n="2">
                  <note pname="b" oct="4" dur="8" dots="1"/>
                </layer>
              </staff>
            </measure>
```

The previous two measure are corrected, but not the last one (unless you have not committed the most recent change).

**Comment by rettinghaus** [comment]
Update: bar lines are now merged, still doubled dot on eight note.
![dotted_notes](https://user-images.githubusercontent.com/7693447/118264907-b8107300-b4b8-11eb-8f9e-44d1a173a9d5.png)

**Comment by Monceber** [comment]
This is fixed by #2306.
